### PR TITLE
Add macosx static build

### PR DESCRIPTION
**Pull request by filonenko-mikhail**
Static image of tarantool is convinient many ways:
- deploy to offline computer using usb sticks, compact disk or even 25 3½″ Floppy disks
- simplified manual: no package managers, no dependency installations

In case when we want to build static image of tarantool for macos

system deps linked dynamically
- libcurl have te be linked dynamically from system
- nghttp2 have to be linked dynamically from system with other name `apple_nghttp2`
- libcurl requires link to -framework Security for ssl features

- libreadline have to linked dynamically with system lib without ncurses

deps linked statically

- libz for openssl have to be find in homebrew ZLIB_ROOT
- libiconv have to be linked statically

result image have to be linked with Security Framework

result binary deps are something like

```
otool -L tarantool
tarantool:
/usr/lib/libedit.3.dylib (compatibility version 2.0.0, current version 3.0.0)
/usr/lib/libcurl.4.dylib (compatibility version 7.0.0, current version 9.0.0)
/usr/lib/libapple_nghttp2.dylib (compatibility version 1.0.0, current version 1.24.1)
/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1252.200.5)
/System/Library/Frameworks/Security.framework/Versions/A/Security (compatibility version 1.0.0, current version 58286.240.4)
/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 400.9.4)
```

### CODE REVIEW

**Comment by Totktonada** [comment]
> libcurl have te be linked dynamically from system

Please, consider #2772.

File: extra/exports [hunk 1]
```diff
@@ -356,3 +356,22 @@ luaJIT_setmode
...
+EVP_CIPHER_block_size
+EVP_CipherInit_ex
+EVP_CipherUpdate
+EVP_CipherFinal_ex
+EVP_DigestInit_ex
+EVP_DigestUpdate
+EVP_DigestFinal_ex
+HMAC_Init_ex
+HMAC_Update
+HMAC_Final
```
Comment (Totktonada) [inline extra/exports]: It is not part of tarantool. Should not be here in a regular build. Please, consider updating of extra/mkexports.

File: cmake/FindReadline.cmake [hunk 2]
```diff
@@ -47,22 +47,22 @@ set(READLINE_INCLUDE_DIRS ${READLINE_INCLUDE_DIR})
...
+    endif()
+    if(EXISTS ${READLINE_INCLUDE_DIR}/readline/rlconf.h)
+        set(CMAKE_REQUIRED_LIBRARIES ${READLINE_LIBRARIES})
+        check_library_exists(${READLINE_LIBRARY} rl_catch_sigwinch ""
+            HAVE_GNU_READLINE)
+        if(HAVE_GNU_READLINE)
+            find_package_message(GNU_READLINE "Detected GNU Readline"
+                "${HAVE_GNU_READLINE}")
+        endif()
+    endif()
```
Comment (Totktonada) [inline cmake/FindReadline.cmake]: Please, avoid such reformatting.

File: cmake/os.cmake [hunk 3]
```diff
@@ -118,6 +118,25 @@ elseif (${CMAKE_SYSTEM_NAME} STREQUAL "Darwin")
...
+        endif()
+        if (NOT DEFINED ZLIB_ROOT)
+            execute_process(COMMAND ${HOMEBREW_EXECUTABLE} --prefix zlib
+                OUTPUT_VARIABLE HOMEBREW_ZLIB
+                OUTPUT_STRIP_TRAILING_WHITESPACE)
+            if (HOMEBREW_ZLIB)
+                message(STATUS "Setting ZLIB root to ${HOMEBREW_ZLIB}")
+                set(ZLIB_ROOT ${HOMEBREW_ZLIB})
+            endif()
+        endif()
```
Comment (Totktonada) [inline cmake/os.cmake]: Should not it be inside FindXXX.cmake?

File: src/main.cc [hunk 4]
```diff
@@ -136,8 +136,10 @@ signal_sigwinch_cb(ev_loop *loop, struct ev_signal *w, int revents)
 	(void) loop;
 	(void) w;
 	(void) revents;
+#if defined(HAVE_GNU_READLINE)
```
Comment (Totktonada) [inline src/main.cc]: What a problem you fix here?

File: .travis.mk [hunk 5]
```diff
@@ -46,7 +46,7 @@ test_ubuntu: deps_ubuntu
 
 deps_osx:
 	brew update
-	brew install openssl readline curl icu4c --force
+	brew install openssl readline curl icu4c libiconv --force
```
Comment (Totktonada) [inline .travis.mk]: Hm. We don't experience any build problems on travis. Why did you added it here?

File: src/main.cc (hunk 4 above)
Comment (filonenko-mikhail) [reply src/main.cc]: Built-in mac readline does not contain `rl_resize_terminal`

**Comment by kostja** [comment]
Please fix the requested changes and resubmit the pull request.

**Pull request closed without merging**